### PR TITLE
Merchant test

### PR DESCRIPTION
**Pull request by codesrobertson**
Adds tests for Edit, Update, and Destroy product controller actions.

### CODE REVIEW

File: app/controllers/products_controller.rb [hunk 1]
```diff
@@ -75,26 +75,37 @@ def edit
   def update
     @product = Product.find_by(id: params[:id])
 
-    if @product.update(product_params)
+    if @product.merchant_id == session[:user_id] && @product.update(product_params)
```
Comment (aecombs) [inline app/controllers/products_controller.rb]: 🤩

File: test/controllers/products_controller_test.rb [hunk 2]
```diff
@@ -69,6 +79,84 @@
...
+    describe "edit" do 
+      it "will retrieve the edit page for an authorized user" do
+        get edit_product_path(@product_test.id)
+
+        must_respond_with :redirect
+        must_redirect_to merchant_products_path(@product_test.id)
+      end 
+
+      it "will redirect to merchant's products path for an unauthorized user" do
+        get edit_product_path(@product_test.id)
```
Comment (aecombs) [inline test/controllers/products_controller_test.rb]: These tests seem incomplete. Will this also redirect for a signed in merchant who doesn't own said product?

File: app/views/merchants/show.html.erb [hunk 3]
```diff
@@ -16,6 +16,7 @@
 <%= link_to "Create new category", new_category_path %>
 <%= link_to "Create new product", new_merchant_product_path(@merchant, @products) %>
 
+<%# Tried Alicia's suggested edit here, but it broke the tests. %>
 <% if @login_merchant && @login_merchant.id == session[:user_id] %>
```
Comment (aecombs) [inline app/views/merchants/show.html.erb:20]: Yeah, I realized later that I misunderstood what you were asking -- I do want to point out that @login_merchant is actually set to session[:user_id] by the filter so this validation doesn't do much other than making sure someone, anyone is logged in. As things are right now, I think any merchant can view any other merchant's dashboard, at least that happened when I was messing around

**Review by cojenco — APPROVED** [review]
Good to go👍 Let's merge!

**Review by aecombs — APPROVED** [review]

**Pull request merged**